core-bundle/src/InsertTag/Resolver/DateInsertTag.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\InsertTag\Resolver;
```

Comment:
I'm not sure if I fully understand the namespace for the core insert tags. Is it because you are effectively "resolving" the specific insert tag you are subscribed to? Maybe a bit confusing as you can also get a `ResolvedInsertTag` instance as the first argument which sounds a bit like there isn't anything to do then. :slightly_smiling_face: 

But maybe it just needs some getting-used-to. :relaxed:

Reply:
The thoughts behind that were like this:

If you parse a string like `{{foo::{{bar}}::baz}}` you get back an instance of `InsertTag`. This can be of the subtype `ResolvedInsertTag` if all nested insert tags are already handled/resolved. Or it can be a `ParsedInsertTag` that gives you access to all nested insert tags too without executing/resolving them.

The `Parsed` and `Resolved` prefixes are used for other classes too that have similar semantics.

An insert tag resolver is a service that gets passed an `InsertTag` and returns an `InsertTagResult`. Other names could be `Handler`, `Replacer`,….

But I’m open to suggestions for a better naming.

Reply:
Thanks for the explainer! Thinking about this again, calling an implementer a `Resolver` makes the most sense to me 👍

Lets keep things as they are 🙂